api/internal/handler/data_loader/route_export.go

```diff
@@ -0,0 +1,441 @@
+	requestBody := &openapi3.RequestBody{}
+	components := &openapi3.Components{}
+	secSchemas := openapi3.SecuritySchemes{}
+	servicePlugins := make(map[string]interface{})
+	plugins := make(map[string]interface{})
+	serviceLabels := make(map[string]string)
+	labels := make(map[string]string)
+
+	for _, route := range routes {
+		extensions := make(map[string]interface{})
+		pathItem := &openapi3.PathItem{}
+		path := openapi3.Operation{}
+		path.Summary = route.Desc
+		path.OperationID = route.Name
+
+		if route.ServiceID != nil {
+			serviceID := utils.InterfaceToString(route.ServiceID)
+			service, err = h.serviceStore.Get(context.Background(), serviceID)
+			if err != nil {
+				if err == data.ErrNotFound {
+					return nil, fmt.Errorf("service id: %s not found", route.ServiceID)
+				}
+				return nil, err
+			}
+
+			_service := service.(*entity.Service)
+			servicePlugins = _service.Plugins
+			serviceLabels = _service.Labels
+		}
+
+		if route.Upstream != nil {
+			extensions["x-apisix-upstream"] = route.Upstream
+		} else if route.UpstreamID != nil && route.Upstream == nil {
+			upstreamID := utils.InterfaceToString(route.UpstreamID)
+			upstream, err := h.upstreamStore.Get(context.Background(), upstreamID)
+			if err != nil {
+				if err == data.ErrNotFound {
+					return nil, fmt.Errorf("upstream id: %s not found", route.UpstreamID)
+				}
+				return nil, err
+			}
+			extensions["x-apisix-upstream"] = upstream
+		} else if route.UpstreamID == nil && route.Upstream == nil && route.ServiceID != nil {
+			_service := service.(*entity.Service)
+			if _service.Upstream != nil {
+				extensions["x-apisix-upstream"] = _service.Upstream
+			} else if _service.Upstream == nil && _service.UpstreamID != nil {
+				upstreamID := utils.InterfaceToString(_service.UpstreamID)
+				upstream, err := h.upstreamStore.Get(context.Background(), upstreamID)
+				if err != nil {
+					if err == data.ErrNotFound {
+						return nil, fmt.Errorf("upstream id: %s not found", _service.UpstreamID)
+					}
+					return nil, err
+				}
+				extensions["x-apisix-upstream"] = upstream
+			}
+		}
+
+		if route.Host != "" {
+			extensions["x-apisix-host"] = route.Host
+		}
+
+		if route.Hosts != nil {
+			extensions["x-apisix-hosts"] = route.Hosts
+		}
+
+		//Parse Labels
+		labels, err = parseLabels(route, serviceLabels, labels)
+		if err != nil {
+			log.Errorf("parseLabels err: ", err)
+			return nil, err
+		}
+
+		if labels != nil {
+			extensions["x-apisix-labels"] = labels
+		}
+
+		if route.RemoteAddr != "" {
+			extensions["x-apisix-remote_addr"] = route.RemoteAddr
+		}
+
+		if route.RemoteAddrs != nil {
+			extensions["x-apisix-remote_addrs"] = route.RemoteAddrs
+		}
+
+		if route.FilterFunc != "" {
+			extensions["x-apisix-filter_func"] = route.FilterFunc
+		}
+
+		if route.Script != nil {
+			extensions["x-apisix-script"] = route.Script
+		}
+
+		if route.ServiceProtocol != "" {
+			extensions["x-apisix-service_protocol"] = route.ServiceProtocol
+		}
```

Comment:
style:  It's better to have a empty line here. Just keep the same style.

Reply:
done.